test/e2e/customconfigs/waitforpodsready_test.go

```diff
@@ -0,0 +1,340 @@
+				var wl kueue.Workload
+				g.Expect(k8sClient.Get(ctx, wlKey, &wl)).Should(gomega.Succeed())
+				g.Expect(wl.Status.RequeueState).ShouldNot(gomega.BeNil())
+				g.Expect(*wl.Status.RequeueState.Count).Should(gomega.Equal(int32(1)))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+	})
+
+	ginkgo.It("should evict and requeue workload when pod failure causes recovery timeout", func() {
+		ginkgo.By("configuring WaitForPodsReady with a long Timeout and a short RecoveryTimeout", func() {
+			cfg := originalCfg.DeepCopy()
+			cfg.WaitForPodsReady = &configapi.WaitForPodsReady{
+				Enable:          true,
+				BlockAdmission:  ptr.To(true),
+				Timeout:         &metav1.Duration{Duration: 5 * time.Minute},
+				RecoveryTimeout: &metav1.Duration{Duration: util.Timeout},
+				RequeuingStrategy: &configapi.RequeuingStrategy{
+					Timestamp:          ptr.To(configapi.EvictionTimestamp),
+					BackoffBaseSeconds: ptr.To(int32(10)),
+					BackoffLimitCount:  ptr.To(int32(1)),
+				},
+			}
+			util.ApplyKueueConfiguration(ctx, k8sClient, cfg)
+			util.RestartKueueController(ctx, k8sClient)
+		})
+
+		ginkgo.By("creating a job", func() {
+			job = testingjob.MakeJob("job-recovery-timeout", ns.Name).
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "2").
+				Parallelism(1).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+		})
+
+		wlKey = types.NamespacedName{
+			Name:      workloadjob.GetWorkloadNameForJob(job.Name, job.UID),
+			Namespace: ns.Name,
+		}
+		workload := &kueue.Workload{}
+		gomega.Eventually(func(g gomega.Gomega) {
+			g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+		var updatedJob *batchv1.Job
+		jobKey := types.NamespacedName{Name: job.Name, Namespace: ns.Name}
+
+		ginkgo.By("simulating the job reporting that its pod is ready", func() {
+			gomega.Expect(k8sClient.Get(ctx, jobKey, job)).Should(gomega.Succeed())
+			updatedJob = job.DeepCopy()
+
+			updatedJob.Status.Active = 1
+			updatedJob.Status.Ready = ptr.To[int32](1)
+			gomega.Expect(k8sClient.Status().Update(ctx, updatedJob)).Should(gomega.Succeed())
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, workload)).Should(gomega.Succeed())
+				cond := apimeta.FindStatusCondition(workload.Status.Conditions, kueue.WorkloadPodsReady)
+				g.Expect(cond).ShouldNot(gomega.BeNil())
+				g.Expect(cond.Status).To(gomega.Equal(metav1.ConditionTrue))
+			}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("simulating a pod failure", func() {
+			gomega.Expect(k8sClient.Get(ctx, jobKey, job)).Should(gomega.Succeed())
+			updatedJob = job.DeepCopy()
+
+			updatedJob.Status.Active = 0
+			updatedJob.Status.Ready = ptr.To[int32](0)
+			updatedJob.Status.Failed = 1
+			gomega.Expect(k8sClient.Status().Update(ctx, updatedJob)).Should(gomega.Succeed())
+		})
+
```

Comment:
Can we check here if workload has PodsReady=false with an intended reason

Reply:
It would be possible, however such a check is not consistent as Kueue often manages to update the status further, causing the test to fail

Reply:
How so, if any of pods failed, then shouldn't the status contain PodsReady=false condition?

Reply:
Introducing `ShortInterval` seems to help, now the test can check for PodsReady=false and not fail (on the scale of 50 tests). Added the check after every simulation of the pod failure

Reply:
What interval would you used before introducing ShortInterval? Shortening the interval in theory shouldn't help as Eventually passes after all conditions are met for the first time

Reply:
Using regular Interval sometimes gave enough time for Kueue to reqeue the workload, so that the check for PodsReady=false would never be satisfied as the status was already changed. With ShortTimeout it seems to be able to catch the time when PodsReady=false with the needed status, allowing the test to move forward

Reply:
I believe the issue was with the cases where the first check is not satisfied and has to wait, when in the meantime Kueue manages to change the status twice. Have removed `ShortInterval` and its corresponding checks, leaving only those that occur pre and post pod failure